### ISSUE TITLE
selector-no-redundant-nesting-selector false positive when followed by a character other than ':'

### ISSUE THREAD

**Issue by gdenisov** [kristerkari/stylelint-scss#32]
'selector-no-redundant-nesting-selector' rule falsely assumes the selector is redundant in this common BEM use case:

```
.block {
  &__element {
    ...
  }
  &--modifier {
   ...
  }
}
```

More generally nesting selector is not redundant if immediately followed by a non white space character as it results in a new not nested rule.

**Comment by kristerkari** [comment]
Thanks for reporting! It looks like we need to add more tests for that rule and refactor the code a bit.
